src/main.cpp

```diff
@@ -115,748 +160,765 @@ int main ( int argc, char** argv )
+            pApplication->addLibraryPath ( QString ( ApplDir.absolutePath() ) );
+        }
+        else
+        {
+            QDir ApplDir ( QCoreApplication::applicationDirPath() );
+            pCoreApplication->addLibraryPath ( QString ( ApplDir.absolutePath() ) );
+        }
 #endif
 
 #if defined( Q_OS_MACX )
-    // On OSX we need to declare an activity to ensure the process doesn't get
-    // throttled by OS level Nap, Sleep, and Thread Priority systems.
-    CActivity activity;
+        // On OSX we need to declare an activity to ensure the process doesn't get
+        // throttled by OS level Nap, Sleep, and Thread Priority systems.
+        CActivity activity;
 
-    activity.BeginActivity();
+        activity.BeginActivity();
 #endif
 
-    // init resources
-    Q_INIT_RESOURCE ( resources );
+        // init resources
+        Q_INIT_RESOURCE ( resources );
 
 #ifndef SERVER_ONLY
-    // clang-format off
+        // clang-format off
 // TEST -> activate the following line to activate the test bench,
 //CTestbench Testbench ( "127.0.0.1", DEFAULT_PORT_NUMBER );
 // clang-format on
 #endif
 
-    CRpcServer* pRpcServer = nullptr;
-
-    if ( iJsonRpcPortNumber != INVALID_PORT )
-    {
-        if ( strJsonRpcSecretFileName.isEmpty() )
+        if ( iJsonRpcPortNumber != INVALID_PORT )
         {
-            qCritical() << qUtf8Printable ( QString ( "- JSON-RPC: --jsonrpcsecretfile is required. Exiting." ) );
-            exit ( 1 );
-        }
+            if ( strJsonRpcSecretFileName.isEmpty() )
+            {
+                throw CErrorExit ( qUtf8Printable ( QString ( "- JSON-RPC: --jsonrpcsecretfile is required. Exiting." ) ) );
+            }
 
-        QFile qfJsonRpcSecretFile ( strJsonRpcSecretFileName );
-        if ( !qfJsonRpcSecretFile.open ( QFile::OpenModeFlag::ReadOnly ) )
-        {
-            qCritical() << qUtf8Printable ( QString ( "- JSON-RPC: Unable to open secret file %1. Exiting." ).arg ( strJsonRpcSecretFileName ) );
-            exit ( 1 );
-        }
-        QTextStream qtsJsonRpcSecretStream ( &qfJsonRpcSecretFile );
-        QString     strJsonRpcSecret = qtsJsonRpcSecretStream.readLine();
-        if ( strJsonRpcSecret.length() < JSON_RPC_MINIMUM_SECRET_LENGTH )
-        {
-            qCritical() << qUtf8Printable ( QString ( "JSON-RPC: Refusing to run with secret of length %1 (required: %2). Exiting." )
-                                                .arg ( strJsonRpcSecret.length() )
-                                                .arg ( JSON_RPC_MINIMUM_SECRET_LENGTH ) );
-            exit ( 1 );
-        }
+            QFile qfJsonRpcSecretFile ( strJsonRpcSecretFileName );
+            if ( !qfJsonRpcSecretFile.open ( QFile::OpenModeFlag::ReadOnly ) )
+            {
+                throw CErrorExit (
+                    qUtf8Printable ( QString ( "- JSON-RPC: Unable to open secret file %1. Exiting." ).arg ( strJsonRpcSecretFileName ) ) );
```

Comment:
Why is a new line needed here? Does clang-format complain otherwise?

Reply:
What new line ? (are you viewing the diff in split mode ?)
The newline after `CErrorExit (` is added by clang-format, the others are due to the split format diff...

Reply:
Yes. Probably that was GitHub...